packages/nvidia_nat_core/src/nat/tool/chat_completion.py

```
@@ -39,39 +46,91 @@ class ChatCompletionConfig(FunctionBaseConfig, name="chat_completion"):
+            usage = Usage(
+                prompt_tokens=prompt_tokens,
+                completion_tokens=completion_tokens,
+                total_tokens=total_tokens,
+            )
+            chat_response = ChatResponse.from_string(output_text, usage=usage)
+
+            if chat_request_or_message.is_string:
+                return GlobalTypeConverter.get().convert(chat_response, to_type=str)
+            return chat_response
 
         except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)